File: ai-service/models/simulation.py
from pydantic import BaseModel
from typing import Optional


class StartupConfig(BaseModel):
    startupId: str
    industry: str
    pricingModel: str
    monthlyBudget: float
    targetAudience: str
    category: str


class SimulationState(BaseModel):
    startup_id: str
    simulation_day: int
    customers: int
    mrr: float
    churn_rate: float
    valuation: float
    runway_months: int
    popularity_score: float
    investor_interest: float
    funding_raised: float
    funding_round: str
    total_revenue: float
    burn_rate: float
    events: list[dict]


class SimulationEvent(BaseModel):
    event_type: str
    title: str
    description: str
    impact: str  # "positive" | "negative" | "neutral"

File: frontend/src/types/simulation.ts
export interface SimulationState {
  id: string;
  startupId: string;
  isRunning: boolean;
  simulationDay: number;
  customers: number;
  mrr: number;
  arr: number;
  churnRate: number;
  valuation: number;
  runwayMonths: number;
  popularityScore: number;
  investorInterest: number;
  fundingRaised: number;
  fundingRound: string;
  totalRevenue: number;
  burnRate: number;
  lastUpdated: string;
}

export interface SimulationMetric {
  id: string;
  startupId: string;
  day: number;
  customers: number;
  mrr: number;
  churnRate: number;
  valuation: number;
  burnRate: number;
  recordedAt: string;
}

export interface SimulationEvent {
  id: string;
  startupId: string;
  eventType: string;
  title: string;
  description: string | null;
  impact: 'positive' | 'negative' | 'neutral';
  metadata: Record<string, unknown>;
  occurredAt: string;
}

export interface Notification {
  id: string;
  title: string;
  message: string;
  type: 'success' | 'warning' | 'error' | 'info';
  timestamp: Date;
}
